tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/CodeownersHelper.cs

```diff
@@ -0,0 +1,512 @@
+using System.Text.RegularExpressions;
+using Azure.Sdk.Tools.Cli.Models.Responses;
+using Azure.Sdk.Tools.CodeownersUtils.Parsing;
+
+namespace Azure.Sdk.Tools.Cli.Helpers
+{
+    public interface ICodeownersHelper
+    {
+        CodeownersEntry? FindMatchingEntries(IList<CodeownersEntry> entries, string path = null, string serviceLabel = null);
+        (CodeownersEntry entry, bool existed) UpdateOrCreateCodeownersEntry(
+            CodeownersEntry? existingEntry,
+            string path,
+            string serviceLabel,
+            List<string> serviceOwners,
+            List<string> sourceOwners,
+            bool isAdding);
+        string NormalizePath(string path);
+        CodeownersEntry findAlphabeticalInsertionPoint(List<CodeownersEntry> codeownersEntries, CodeownersEntry codeownersEntry);
+        public (List<CodeownersEntry>, int) mergeCodeownerEntries(List<CodeownersEntry> codeownersEntries, int index);
+        string addCodeownersEntryAtIndex(string codeownersContent, CodeownersEntry codeownersEntry, int index, bool codeownersEntryExists);
+        string formatCodeownersEntry(CodeownersEntry codeownersEntry);
+        (int StartLine, int EndLine) findBlock(string currentContent, string serviceCategory);
+        public string ExtractDirectoryName(string path);
```

Comment:
```suggestion
        string ExtractDirectoryName(string path);
```

`public` is implied in interfaces